src/coap_blockwise.c

```diff
@@ -103,6 +99,19 @@ static int call_read_block_callback(struct blockwise_transfer *ctx)
     return block_size;
 }
 
+// Convert index values between block sizes
+static int recalculate_block_idx(int from_idx, size_t from_larger_szx, size_t to_smaller_szx)
+{
+    assert(from_larger_szx >= to_smaller_szx);
+
+    /* Recalculate index so what was sent is now based on the new block_size */
+    int blocks_sent = from_idx + 1;
+    int size_change_multiplier = (1 << (from_larger_szx - to_smaller_szx));
+
+    /* Subtract 1 to indicate the block we just sent; inc happens after return */
```

Comment:
This function shouldn't assume what the caller will or will not do. Instead, it should describe what it does, and the caller can act accordingly.